src/main/java/org/fogbowcloud/saps/engine/core/archiver/storage/swift/SwiftPermanentStorage.java

```diff
@@ -86,20 +103,18 @@ private boolean checkPropertiesDebugMode(Properties properties) {
      * This function tries to archive a task trying each folder in order
      * (inputdownloading -> preprocessing -> processing).
      *
-     * @param task task to be archived
+     * @param taskId The Task's unique identifier
      * @return boolean representation, success (true) or failure (false) in to
      * archive the three folders.
      */
     @Override
-    public boolean archive(SapsImage task) {
-        String taskId = task.getTaskId();
-
+    public boolean archive(String taskId, boolean isFailed) {
         LOGGER.info("Attempting to archive task [" + taskId + "] with a maximum of " + MAX_ARCHIVE_TRIES
                 + " archiving attempts for each folder (inputdownloading, preprocessing, processing)");
 
-        String swiftExports = (task.getState() == ImageTaskState.FAILED && this.debugMode)
-                ? properties.getProperty(SapsPropertiesConstants.PERMANENT_STORAGE_DEBUG_TASKS_DIR)
-                : properties.getProperty(SapsPropertiesConstants.PERMANENT_STORAGE_TASKS_DIR);
+        String swiftExports = (isFailed && this.debugMode)
+                ? debugTasksDirName
+                : tasksDirName;
```

Comment:
lines [119, 126] ask for a private method. I'm ok to drop a FIXME and improve it latter

Reply:
Done.